### PR TITLE
test(gatsby-plugin-image): add test for gatsby-plugin-image browser

### PR DESCRIPTION
**Pull request by mfrachet**
## Description

Added some tests for gatsby-image.browser

Screenshot 2020-11-16 at 15 59 44

I did what I had in mind, this was quite tough to tests 😅 

Feel free to ask for more, I'm maybe missing some cases 😬

### CODE REVIEW

File: packages/gatsby-plugin-image/src/components/__tests__/gatsby-image.browser.tsx [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+      height: 100,
+      layout: `fluid`,
+      images: { fallback: { src: `some-src-fallback.jpg` } },
+      placeholder: { sources: [] },
+      sizes: `192x192`,
+      backgroundColor: `red`,
+    }
+
+    beforeHydrationContent = document.createElement(`div`)
+    beforeHydrationContent.innerHTML = `
```
Comment (gillkyle) [inline packages/gatsby-plugin-image/src/components/__tests__/gatsby-image.browser.tsx:37]: Woah, is this to replicate the SSR? Clever.

**Review by gillkyle — APPROVED** [review]
Since me and @mfrachet pulled up this code together and walked through the logic in `gatsby-image.browser.tsx` and these tests side by side I'm inclined to say this looks good! I learned a thing or two about some testing utilities along the way too 🙂

**Review by ascorbic — APPROVED** [review]
This is awesome. Great work, @mfrachet ! I've not tested hydration before. Really interesting to see how it's done, and I feel a lot more confident with this in.

File: packages/gatsby-plugin-image/src/components/__tests__/gatsby-image.browser.tsx (hunk 1 above)
Comment (ascorbic) [reply packages/gatsby-plugin-image/src/components/__tests__/gatsby-image.browser.tsx:37]: Nice

**Comment by mfrachet** [comment]
Happy it increases confidence! Thanks for the kind words 😊

**Pull request merged**